File: Metode-Avansate-de-Programare/lab2/decorator/PrinterTaskRunner.java
package sem1_2.decorator;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class PrinterTaskRunner extends AbstractTaskRunner {

    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");

    public PrinterTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        super.executeOneTask();
        System.out.println("Task executed at: " + LocalDateTime.now().format(TIME_FORMATTER));
    }
}

File: Metode-Avansate-de-Programare/lab11/lab11/PrinterTaskRunner.cs
using System;
using System.Globalization;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab11;

public class PrinterTaskRunner : AbstractTaskRunner
{
    private static readonly string TimeFormat = "HH:mm";

    public PrinterTaskRunner(TaskRunner taskRunner) : base(taskRunner)
    {
    }

    public override void executeOneTask()
    {
        base.executeOneTask();
        Console.WriteLine($"Task executed at: {DateTime.Now.ToString(TimeFormat, CultureInfo.InvariantCulture)}");
    }
}